src/common/Elsa.Mediator/Middleware/Command/CommandPipelineBuilder.cs

```diff
@@ -38,8 +38,10 @@ public CommandMiddlewareDelegate Build()
     {
         CommandMiddlewareDelegate pipeline = _ => new ValueTask();
 
-        foreach (var component in _components.Reverse())
-            pipeline = component(pipeline);
+        for (int i = _components.Count - 1; i >= 0; i--)
+        {
+            pipeline = _components[i](pipeline);
+        }
```

Comment:
This changes seems both more complex and less performant (theoretically speaking), given that now we're using am indexer into the components list during each iteration. What's the upshot to this change that I don't see?

Reply:
Well this is purely because `.Reverse()` now maps to 'List<T>.Reverse()' which actually reverses the list data and is a `void` method 🙂